contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/GobblinDistcpJob.java

```diff
@@ -0,0 +1,217 @@
+package voldemort.store.readonly.mr.azkaban;
+
+import java.util.Map;
+import java.util.List;
+import java.net.URI;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.permission.FsAction;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.apache.hadoop.mapred.JobConf;
+
+import gobblin.runtime.api.JobExecutionResult;
+import gobblin.runtime.embedded.EmbeddedGobblin;
+import gobblin.runtime.embedded.EmbeddedGobblinDistcp;
+
+import voldemort.utils.Props;
+import voldemort.utils.logging.PrefixedLogger;
+
+import azkaban.jobExecutor.AbstractJob;
+
+
+public class GobblinDistcpJob extends AbstractJob {
+    private final static String ATTR_PREFIX = "distcpConf.";
+    private String source;
+    private final String destination;
+    private final Props props;
+
+    GobblinDistcpJob(String id, String sourceHdfsCluster, String destinationVoldemortCluster, Props props) {
+        super(id, PrefixedLogger.getLogger(GobblinDistcpJob.class.getName(), destinationVoldemortCluster));
+        this.source = sourceHdfsCluster;
+        this.destination = destinationVoldemortCluster;
+        this.props = props;
+    }
+
+    public void run() throws Exception {
+        info("###############  Distcp  ###############");
+        try {
+            String cdnURL = pickCDN();
+            if (!cdnURL.isEmpty()) {
+                info("Using CDN: " + cdnURL);
+                String username = props.getString("env.USER", "unknownUser");
+                String pathPrefix = props.getString(VoldemortBuildAndPushJob.PUSH_CDN_PREFIX).trim();
+                assert pathPrefix.startsWith("/");
+                pathPrefix = removeTrailingSlash(pathPrefix);
+                // Replace original cluster with CDN, e.g. hdfs://original:9000/a/b/c => webhdfs://cdn:50070/prefix/user/a/b/c
+                String cdnDir =  cdnURL + pathPrefix + "/" + username + extractPathFromUrl(source);
```

Comment:
If the distcp will be execute with each individual user, what will be the file permission in destination cdn?
From my understanding, it should be 770, so that the users inside the same group will be able to run the same store push job.